### DISCUSSION TITLE
How to use string array values as object keys?

### DISCUSSION THREAD

**Discussion by navtoj** [colinhacks/zod#1867] [Q&A] [answered]
I have a list of strings: `const filenames = z.enum(['one', 'two', 'three']);`

..and want use those values as the possible keys of an object: 

```ts
const hashes = z.object({
    [filenames]: z.string().length(40)
});
```

Is something like that possible is Zod? Any suggestions?

p.s. I know going to other way is possible with [`z.object({}).keyof()`](https://zod.dev/?id=keyof) but that's not what i'm looking for.

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you're looking for?

https://github.com/colinhacks/zod#records

```ts
const filenames = z.enum( [ 'one', 'two', 'three' ] )
const hashes = z.record( filenames, z.string().length( 40 ) )
type Hashes = z.infer<typeof hashes>
// type Hashes = {
//     one?: string | undefined
//     two?: string | undefined
//     three?: string | undefined
// }
```

**Reply by navtoj** [reply]
@JacobWeisenburger Also, whats the difference between `.Enum` and `.Values`?
![image](https://user-images.githubusercontent.com/20136585/212256998-35d53860-3302-4ca0-b64c-d190adb3a2f7.png)

I'm using the enum values for simple variable comparison and both `filenames.Enum.one` and `filenames.Values.one` seem to return the same thing.
![image](https://user-images.githubusercontent.com/20136585/212257820-85c70641-5d29-40de-be7e-1e5b6e20941d.png)

**Reply by JacobWeisenburger** [reply]
I think they are the same. Not sure why though.

**Reply by navtoj** [reply]
Yea, they return the exact same thing. Maybe for backward compatibility reasons?
https://github.com/colinhacks/zod/blob/33c771d88281429dcb53a4a33d183fa9eb6a8261/src/types.ts#L3582-L3604
